tests/components/thethingsnetwork/test_init.py

```diff
@@ -0,0 +1,78 @@
+"""Define tests for the The Things Network init."""
+
+import pytest
+from ttn_client import TTNAuthError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import (
+    device_registry as dr,
+    entity_registry as er,
+    issue_registry as ir,
+)
+from homeassistant.setup import async_setup_component
+
+from .conftest import (
+    APP_ID,
+    CONFIG_ENTRY,
+    DATA_UPDATE,
+    DEVICE_FIELD,
+    DEVICE_FIELD_2,
+    DEVICE_ID,
+    DEVICE_ID_2,
+    DOMAIN,
+)
+
+
+async def test_normal(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    device_registry: dr.DeviceRegistry,
+    mock_TTNClient_coordinator,
+) -> None:
+    """Test a working configurations."""
+    CONFIG_ENTRY.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(CONFIG_ENTRY.entry_id)
+
+    await hass.async_block_till_done()
+
+    # Check devices
+    assert (
+        device_registry.async_get_device(
+            identifiers={(DOMAIN, f"{APP_ID}_{DEVICE_ID}")}
+        ).name
+        == DEVICE_ID
+    )
+
+    # Check entities
+    assert entity_registry.async_get(f"sensor.{DEVICE_ID}_{DEVICE_FIELD}")
+
+    # Test reaction to options update
+    hass.config_entries.async_update_entry(
+        CONFIG_ENTRY, data=CONFIG_ENTRY.data, options={"dummy": "new_value"}
+    )
+
+    assert not entity_registry.async_get(f"sensor.{DEVICE_ID_2}_{DEVICE_FIELD}")
+    push_callback = mock_TTNClient_coordinator.call_args.kwargs["push_callback"]
+    await push_callback(DATA_UPDATE)
+    assert entity_registry.async_get(f"sensor.{DEVICE_ID_2}_{DEVICE_FIELD_2}")
```

Comment:
Maybe these tests would better suit a `test_sensor.py` and then just keep the `test_init.py` for testing about the initialization of the integration

Reply:
This is testing the coordinator when not able to coonect. Movving to test_coordinator.py

Reply:
But you're not testing the coordinator, you're testing what not able to connect is effectively doing to the entities that depend on it. I dislike coordinator tests more tbf because they expect internals, even though we test the functionality

Reply:
ok, I see your point - moved the test_normal to test_sensor.py